stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewHighlightOverlays.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.elements.android;
+
+import android.annotation.TargetApi;
+import android.graphics.Canvas;
+import android.graphics.Rect;
+import android.graphics.Region;
+import android.graphics.drawable.ColorDrawable;
+import android.os.Build;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewGroup.MarginLayoutParams;
+
+abstract class ViewHighlightOverlays {
+
+  abstract void highlightView(View view, int mainColor);
+
+  abstract void removeHighlight(View view);
+
+  static ViewHighlightOverlays newInstance() {
+    // This may not be needed since ViewHighlighter.newInstance() is already instantiating a
+    // NoopHighlighter for SDK_INT < JELLY_BEAN_MR2, but just to make sure...
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return new ViewHighlightOverlaysJellybeanMR2();
+    }
+    return new NoOpViewHighlightOverlays();
+  }
+
+  private static class NoOpViewHighlightOverlays extends ViewHighlightOverlays {
+
+    @Override
+    void highlightView(View view, int mainColor) {
+    }
+
+    @Override
+    void removeHighlight(View view) {
+    }
+  }
+
+  @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+  private static class ViewHighlightOverlaysJellybeanMR2 extends ViewHighlightOverlays {
+    private static final int MARGIN_OVERLAY_COLOR = 0xaaf4ca9e;
+    private static final int PADDING_OVERLAY_COLOR = 0xaabedab6;
+    private final HighlightDrawable[] mHighlightDrawables = {
+        new MainHighlightDrawable(),
+        new PaddingTopHighlightDrawable(),
+        new PaddingBottomHighlightDrawable(),
+        new PaddingRightHighlightDrawable(),
+        new PaddingLeftHighlightDrawable(),
+        new MarginTopHighlightDrawable(),
+        new MarginBottomHighlightDrawable(),
+        new MarginRightHighlightDrawable(),
+        new MarginLeftHighlightDrawable()
+    };
+
+    ViewHighlightOverlaysJellybeanMR2() {
+    }
+
+    @Override
+    void highlightView(View view, int mainColor) {
+      mHighlightDrawables[0].setColor(mainColor);
```

Comment:
This is a fragile way to address the `MainHighlightDrawable`. Someone could change the contents or ordering of the array and not even notice that they're breaking this. We should put that into its own field and access it directly, e.g.
```
private final MainHighlightDrawable mMainHighlightDrawable = new MainHighlightDrawable();
private final HighlightDrawable[] mHighlightDrawables = {
    mMainHighlightDrawable,
    new PaddingTopHighlightDrawable(),
    ...
```